HucaresServer/src/HucaresServer.DataAcquisition.UnitTests/CameraInfoDownloadingTests.cs

```diff
@@ -1,11 +1,101 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using FakeItEasy;
+using HucaresServer.Storage.Helpers;
+using HucaresServer.Storage.Models;
 using NUnit.Framework;
+using Shouldly;
 
 namespace HucaresServer.DataAcquisition.UnitTests
 {
     [TestFixture]
     public class CameraInfoDownloadingTests
     {
+        [Test]
+        public void DownloadImagesFromCameraInfoSources_WithTrusted_ShouldDownloadAndReturn()
+        {
+            // Arrange
+            var fakeBitmap = new Bitmap(100, 100);
+            var url = "https://some.url";
+            
+            ImageConverter converter = new ImageConverter();
+            var fakeImageArray =  (byte[])converter.ConvertTo(fakeBitmap, typeof(byte[]));
+            
+            
+            var fakeWebClient = A.Fake<IWebClient>();
+            A.CallTo(() => fakeWebClient.DownloadData(url))
+                .Returns(fakeImageArray);
+            
+            var fakeWebClientFactory = A.Fake<IWebClientFactory>();
+            A.CallTo(() => fakeWebClientFactory.BuildWebClient())
+                .Returns(fakeWebClient);
+            
+            var fakeMissingPlates = new List<CameraInfo>()
+            {
+                new CameraInfo() {Id = 0, HostUrl = url, IsTrustedSource = true},
```

Comment:
Add a non trusted source to ensure only 1 gets picked.

Reply:
It is actually CameraInfoHelper responsibility to filter the values by IsTrustedSource, and here I fake the helper and call to it, so I have to return already filtered data.